### PR TITLE
⚡️  NICE-58 performance updates

### PR DESCRIPTION
**Pull request by JeromeFitz**
Would like to finally move a parallel website outside of this repo into here, heh:

## Commits

- ⚡️  dont use esm import for radix-ui [b]
- ⚡️  i actually think this hurts perf now ☉_☉ [b]
- ⚡️  cx (!twMerge); 404 link home 
- ⚡️  (next) dynamic !ssr is not ... well working
- ⚡️  siteColors; next/dynamic; cx-vs-twMerge
- 📌  (deps) `@types.node@18.11.9`
- ♻️  moving files around again
- 🚨  (tailwind) tap-dance around siteColors narrowed list
- ⚡️  tsup build; lhci improvements; tailwind
- ⚡️  icon first visually, second in html
- ⚡️  (notion) !EmojiWrapper for now
  - Aye, since `notion` is pretty much _the reason_ for the website, delaying the imports for `node-emoji|emoji-regex` ends up doing nothing but causing more threads to happen.
  - Turning off for now. Instead should pass a configuration setting if people want this or not.
  - I say yes, though descriptors leave a lot to be desired, especially as visually it is better with spacing
  - Acceptable performance hit/trade-off in my estimation. But geez, that was hard to figure out, haha.
- 🐛  (next) Image.client, and optimizations
  - Pass `sizes` 😑 haha how was this also turned off?
- 🚚  (next) static is redundant, + favicon/manifest
  - No need for PWA but some things are nice
- 🔐  (next) _privateFolders
- 🔧  (tsconfig) shared for dev|prod split due to aggressive next
- ⚡️  (next) preconnects, final meta stuff, ship 🚀
  - Ah yes, forgot the `preconnect` hack 🤣
  - Added some more meta stuff too while we were here
  - Altered some sizing on large screens
- 👷  (e2e) npx not pnpm for random installs
  - This should not break `pnpm|turbo` cache -- right? 🙏🏻 

## Breakdown

- Site Architecture movement
- `next/dynamic` should be sparesly used I think
  - If SSR _can_ load it, and it is not a heavy `css|js` hit, this would degrade non-js browsers
- `Emoji` This is not the way to do this, heh
  - yes `node-emoji` (`41.1kB`) and `emoji-regex` (`28.5kB`) are hits
  - but they serve a repetitive purpose _throughout_
  - this visual to html swap works but requires some trade-offs that I do not like
  - need to officially confirm about wrapping emojis with `alt` text too
- `twMerge` is a `~6kB` hit, but it does a lot of _good_
  -  Went through and made everything `cx` official
  - Only 1-2 needed `twMerge` still -- and can be done
- Move away from `framer-motion` until we can determine actual site use
  - `Loading` is not enough of a warranted reason
- Unused `css` from Radix-UI imports
  - On Development we go very dynamic in ensuring everything is available to us
  - For Production we should limit the amount of `css` where possible
  - This is not the _best_ way to go about this as it is confusing, but on the right track at least  
  - Need to tap-dance around dynamic imports and the _now_ excluded colors

## Stats

Good news, we are starting from a Vercel RES of `100` on everything

The thing to hyper-focus on at the moment is Google:
- Lighthouse (`pnpm lhci` w/ PR stats) & PageSpeed Insights (`https://pagespeed.web.dev/`)

**Before:**
- Performance: 92
- Accessibility: 97
- Best Practices: 100
- SEO: 100

`*` Accessibility:   This is/was due to Radix-UI Pills where we follow their lead on 4 BG 11 Text
  - Passes their tests, not Google
  - To appease we drop to 3 BG (though should be 4 BG on Light)

**After:**
- Performance: 98
- Accessibility: 100
- Best Practices: 100
- SEO: 100

## Next Steps

Final port of un-shared stuff -- though we can probably just transpile or way past this short-term

SEO
  - Misleading as we need to plant-based beef up the Next Metadata generation
  - Recommendation: `/${slug}/seo` and continue to keep separate in KV
  - Also KV is probably not ideal and should we move to Postgres? TOO MNAY QUESTIONS
    - Upstash has no problems, but if we did Vercel we would hit their limits (through Upstash ... 🥴)
    
    
 ## Outcome

Aye, I'll take it. We have to do a `skip ci` to put `@main` back of the gh-action workflows:

image

image

image

### CODE REVIEW

File: packages/tailwind-config/src/backgrounds.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ radixColors.map((_color) => {
     color: `var(--${color}11)`,
   }
   backgrounds['.' + color + '-cta-int'] = {
-    backgroundColor: `var(--${color}4)`,
+    backgroundColor: `var(--${color}34`,
```
Comment (JeromeFitz) [inline packages/tailwind-config/src/backgrounds.js:34]: Uh wut. `3` dood

File: sites/jeromefitzgerald.com/src/components/Notion/Blocks/Image.tsx [hunk 2]
```diff
@@ -216,5 +215,6 @@ async function ImageImpl({
   )
 }
 
+export { Image as NextImage }
```
Comment (JeromeFitz) [inline sites/jeromefitzgerald.com/src/components/Notion/Blocks/Image.tsx:218]: 🥴 This makes my head hurt a bit.

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>b116dff</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-cbnowkt6v-jerome5.vercel.app'>https://jeromefitzgerald-cbnowkt6v-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/DAdpKyKBpb5zzcpVBcDwZebXR3hf'>https://vercel.com/jerome/jeromefitzgerald/DAdpKyKBpb5zzcpVBcDwZebXR3hf</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5501535158)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>79414bb</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-apde9k40b-jerome5.vercel.app'>https://jeromefitzgerald-apde9k40b-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/D41PbaHBCDMNPMusSStVv9fMeyth'>https://vercel.com/jerome/jeromefitzgerald/D41PbaHBCDMNPMusSStVv9fMeyth</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5501728863)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>18b1a89</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-438m4uzgf-jerome5.vercel.app'>https://jeromefitzgerald-438m4uzgf-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/Ajemv8EDJr2bTkFQ6ZNrccQt6oYG'>https://vercel.com/jerome/jeromefitzgerald/Ajemv8EDJr2bTkFQ6ZNrccQt6oYG</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5501850716)

File: sites/jeromefitzgerald.com/src/components/Notion/Blocks/Image.tsx (hunk 2 above)
Comment (JeromeFitz) [reply sites/jeromefitzgerald.com/src/components/Notion/Blocks/Image.tsx]: Well. Still hurts, but okay. `Image.client` does not seem as bad as that is more of a wrapper.

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>8bf9ac3</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-o13h37jpt-jerome5.vercel.app'>https://jeromefitzgerald-o13h37jpt-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/ANLWUDund81JnmpEd7PBBexcMpJU'>https://vercel.com/jerome/jeromefitzgerald/ANLWUDund81JnmpEd7PBBexcMpJU</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5502045480)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>033f818</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-dz0vmcwbp-jerome5.vercel.app'>https://jeromefitzgerald-dz0vmcwbp-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/2AVC89eH6cEss9wxyPWoBA8KeHPL'>https://vercel.com/jerome/jeromefitzgerald/2AVC89eH6cEss9wxyPWoBA8KeHPL</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5502712235)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>b19015a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-198uf0c1e-jerome5.vercel.app'>https://jeromefitzgerald-198uf0c1e-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/3wtvHvRoB8AosWd3dnDq4McwevTn'>https://vercel.com/jerome/jeromefitzgerald/3wtvHvRoB8AosWd3dnDq4McwevTn</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5502907471)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>d01f670</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-4squnkxv1-jerome5.vercel.app'>https://jeromefitzgerald-4squnkxv1-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/6hmn9jhYAiAvDJkhPubhHJ73DGtn'>https://vercel.com/jerome/jeromefitzgerald/6hmn9jhYAiAvDJkhPubhHJ73DGtn</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5503040088)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>318889a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-li1a9bg7m-jerome5.vercel.app'>https://jeromefitzgerald-li1a9bg7m-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/H9MySuiqNYKYSQzPXPwRhjSmSmXU'>https://vercel.com/jerome/jeromefitzgerald/H9MySuiqNYKYSQzPXPwRhjSmSmXU</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5503291604)

**Comment by BotJerome** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>b8e7ff7</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://jeromefitzgerald-ilfzsexyf-jerome5.vercel.app'>https://jeromefitzgerald-ilfzsexyf-jerome5.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/jerome/jeromefitzgerald/5pdMJGvxkFQ4jkXGCb4pHcRBrhWK'>https://vercel.com/jerome/jeromefitzgerald/5pdMJGvxkFQ4jkXGCb4pHcRBrhWK</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/JeromeFitz/websites/actions/runs/5503483083)

**Pull request merged**